libr/util/str.c

```diff
@@ -3583,6 +3656,37 @@ R_API const char *r_str_str_xy(const char *s, const char *word, const char *prev
 	return d;
 }
 
+R_API size_t r_charset_encode_str(ut8 *asciistr, ut8 *in, size_t len_input, RCharset *r_char) {
+	bool is_in_set;
+	size_t j, k;
+	memset(asciistr, 0, len_input);
+	ut8 * buff_hex = (ut8 *) malloc (len_input);
+
+	for (j = 0; j < len_input; j ++) {
+		is_in_set = true;
+		buff_hex[0] = in[j];
+		buff_hex[1] = '\0';
+
+		for (k = 0; search_from_hex (r_char->custom_charset, buff_hex) == NULL && k < (len_input-j); k ++) {
+			buff_hex[k] = in[j+k];
+			if (k >= (len_input-j)-1) {
+				is_in_set = false;
+			}
+		}
+		buff_hex[k+1] = '\0';
+
+
+		if (!is_in_set) {
+			break;
+		}
+		r_str_cat ((char *) asciistr, (char *) search_from_hex (r_char->custom_charset, buff_hex)->ch);
```

Comment:
searcH-from_hex can return null, better move this one line up with a separate variable to not make this statement too long and chk for null